docs/development/commit-messages.md

```diff
@@ -0,0 +1,160 @@
+category: development
+tags: [development, commit, git]
+order: 1
+---
+
+# Commit Messages
+
+Commit messages are an incredibly important part of development, as they provide
+a summary of the changes that are included in a commit. This is instrumental in
+producing a clear history for a given project.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Where these Guidelines Apply](#where-these-guidelines-apply)
+- [Formatting Your Commit Messages](#formatting-your-commit-messages)
+  - [Summary Line](#summary-line)
+  - [Detailed Explanation](#detailed-explanation)
+  - [Issue Number](#issue-number)
+  - [General Formatting](#general-formatting)
+
+<!-- /MarkdownTOC -->
+
+## Where these Guidelines Apply
+
+These guidelines apply to the following branches in the Enso repository:
+
+- The main branch (`main`).
+- Release branches (those beginning with `release/`, as specified in the
+  [release policy](../distribution/release-policy.md#release-branches)).
+
+This means that you _must_ be careful to apply these guidelines when creating
+the squash commit to merge your PRs into `main`, as that commit message must
+follow the guidelines.
+
+## Formatting Your Commit Messages
+
+Your commit messages should be formatted as follows:
+
+```plain
+[Type] A brief summary line
+
+1.  A numbered list.
+2.  That describes the changes made.
+
+Related #issue-no
+```
+
+All elements are mandatory for all commits. Please note that the blank line
+between the summary and the numbered list, and the line between the numbered
+list and the `Resolves` are crucial to prevent certain tools from getting
+confused.
+
+An example of a good commit message is as follows:
+
+```plain
+[Fix, Style] Fix a crash in the frobnicator
+
+1.  This commit fixes a bug in the method resolver that had arisen due
+    to an issue with the frobnicator. Tweaking the algorithm used by
+    the frobnicator from super-fancy-algo to even-fancier-algo resolved
+    the issue.
+2.  It also fixes a minor style issue with the naming of builder
+    methods for the Truffle nodes.
+
+Resolves #211
+Resolves #283
+```
+
+> #### On GitHub
+>
+> When merging a PR on GitHub you are given _two_ boxes. The small, single-line
+> one is where the summary line goes. The larger box should contain the
+> remaining content.
+
+### Summary Line
+
+The summary line is a short descriptor of the commit. It must meet the following
+requirements:
+
+1.  It is written as an imperative "Create a new API for foo", not "Creates a
+    new API for foo".
+2.  It is <= 50 characters long.
+3.  It begins with a capital letter.
+4.  It _does not_ end with a full stop (`.`).
+5.  It should specify the commit type(s) from the list of
+    [commit types](#commit-types) in square brackets (`[]`) at the start.
+6.  It does not contain spurious capitalisation. It isn't a title (and so should
+    not be in title case), but is instead a summary. Capitalise proper nouns and
+    the first word, but that's it.
+
```

Comment:
I feel like summary line should also contain number of PR it is coming from

Reply:
This is included in the body instead. It's visual noise in the summary line imo.

Reply:
The current description only mentions the related issue, but not the PR that this commit is associated with. Maybe that should be added to the description?